pkg/authorization/admission/restrictusers/subjectchecker.go

```diff
@@ -0,0 +1,247 @@
+package restrictusers
+
+import (
+	"fmt"
+
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	oclient "github.com/openshift/origin/pkg/client"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// SubjectChecker determines whether rolebindings on a subject (user, group, or
+// service account) are allowed in a namespace.
+type SubjectChecker interface {
```

Comment:
I'm expecting unit tests for individual subject checkers, then unit tests for the admission plugin focus on "did I describe the deltas correctly and run all the checks in the chain"